### PR TITLE
Application Gateway - Adds Rewrite rule set for Header CRUD

### PR DESCRIPTION
**Pull request by sai-manoj-kumar**
Adds the structure of rewriteRuleSets in the applicationGatewayProperties.
Adds recursively the sub structures in the rewriteRuleSet.
Adds the reference to rewriteRuleSet in the requestRoutingRule for both Basic rule and Path based rule.

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by MikhailTryakhov** [comment]
do we need to create any examples?

**Comment by sai-manoj-kumar** [comment]
> 
> 
> do we need to create any examples?
@MikhailTryakhov , I have added the examples.

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2018-10-01/applicationGateway.json [hunk 1]
```diff
@@ -1474,6 +1482,86 @@
       ],
       "description": "Request routing rule of an application gateway."
     },
+    "ApplicationGatewayRewriteRuleSet": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ApplicationGatewayRewriteRuleSetPropertiesFormat"
+        },
+        "name": {
```
[the remark this answers is not in the document]
Comment (sai-manoj-kumar) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-10-01/applicationGateway.json]: etag is also readOnly. Marked it also as readOnly. No other readOnly properties present.

**Review by dsgouda — APPROVED** [review]
LGTM, will merge when CIs pass. By passing ARM review since this is high pri

**Comment by sai-manoj-kumar** [comment]
Checks have passed but it is not merged automatically. Should we merge it manually?

**Pull request merged**